source/plugin/Assets/GoogleMobileAds/Platforms/Unity/BannerClient

```diff
@@ -0,0 +1,238 @@
+// Copyright (C) 2020 Google LLC.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Reflection;
+using GoogleMobileAds.Api;
+using GoogleMobileAds.Common;
+
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace GoogleMobileAds.Unity
+{
+    public class BannerClient : BaseAdDummyClient, IBannerClient
+    {
+        // Ad event fired when the banner ad has been received.
+        public event EventHandler<EventArgs> OnAdLoaded;
+        // Ad event fired when the banner ad has failed to load.
+        public event EventHandler<AdFailedToLoadEventArgs> OnAdFailedToLoad;
+        // Ad event fired when the banner ad is opened.
+        public event EventHandler<EventArgs> OnAdOpening;
+        // Ad event fired when the banner ad is closed.
+        public event EventHandler<EventArgs> OnAdClosed;
+        // Ad event fired when the banner ad is leaving the application.
+        public event EventHandler<EventArgs> OnAdLeavingApplication;
+        // Ad event fired when the banner ad is estimated to have earned money.
+        public event EventHandler<AdValueEventArgs> OnPaidEvent;
+
+        public GameObject prefabAd, dummyAd;
+
+        private Dictionary<AdSize, string> prefabAds = new Dictionary<AdSize, string>()
+        {
+            {AdSize.Banner, "DummyAds/Banners/BANNER"},
+            {AdSize.SmartBanner, "DummyAds/Banners/SMART_BANNER" },
+            {AdSize.MediumRectangle, "DummyAds/Banners/MEDIUM_RECTANGLE" },
+            {AdSize.IABBanner, "DummyAds/Banners/FULL_BANNER" },
+            {AdSize.Leaderboard, "DummyAds/Banners/LEADERBOARD" },
+            {new AdSize (320,100), "DummyAds/Banners/LARGE_BANNER" }
+        };
+        
+        public BannerClient()
```

Comment:
not essential now but i imagine this constructor and the common properties (prefab ad, dummy ad) will probably go in the base class for re-use in other formats.